### ISSUE TITLE
Server side check for YouTube links

### ISSUE THREAD

**Issue by p2r3** [p2r3/epochtal#12]
Currently, only the frontend does a basic substring check to verify YouTube videos. The server should do a similar check, along with a query to the YouTube API ensuring that the video is unlisted and within some range of the specified run time.

**Comment by PancakeTAS** [comment]
I'd also like to chime in with my opinion on this here. In my opinion, only allowing the submission links to be YouTube links is a bit too limiting, as some people may want to use another video host or perhaps not even submit a video at all and rather a perhaps website (to give an example, translate% would benefit from being hosted on another webpage specifically made for these video-style text adventures.). Or perhaps, the YouTube bitrate just destroys the video quality too much, considering it's again being re-encoded by the livestream host.

What I would recommend is this: yt-dlp supports basically any website you throw at it, from generic mp4 videos to self made websites with video tags on them, it will always be able to grab the video file and in most cases even length (though length does not always work). Epochtal should easily be able to deduce the length of a submission or else qualify it as "other" if it isn't a video submission.

Additionally this would come with the benefit of Epochtal not being dependent on YouTube and would make the system more standalone.

**Comment by p2r3** [comment]
True, using yt-dlp does seem like a better choice. However, I remember it having issues with querying data from age restricted videos? I suppose the YouTube API could be used for those exclusively. It _is_ kind of weird to mix different approaches like this, but it'd be even worse not to enforce consistent rules across all submissions (like demanding that the submission time roughly matches the video time).

**Comment by PizzaPost** [comment]
> Currently, only the frontend does a basic substring check to verify YouTube videos. The server should do a similar check, along with a query to the YouTube API ensuring that the video is unlisted and within some range of the specified run time.

It also should check if it’s a playlist cause I can submit it to the website. That has a big problem for the 1h limit and the submission end.
Example:
I create a playlist with a video and than add another one and delete the first. So the new one could be recorded after the submission end or weeks ago.
